### PR TITLE
[PM-5571] Migrate enableDDG to state provider framework

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Migrate enableDuckDuckGo to state provider framework.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/desktop/src/main/native-messaging.main.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class NativeMessagingMain {
     private exePath: string,
   ) {}
 
-  async listen() {
```
Comment (Hinton) [inline apps/desktop/src/main/native-messaging.main.ts:27]: This method didn't need to be async, and removing it let us get rid of the eslint warnings when calling.

**Review by cagonzalezcs — DISMISSED** [review]
Looks clean and straightforward from my side :)

File: apps/desktop/src/autofill/services/desktop-autofill-settings.service.ts
Comment (jprusik) [inline apps/desktop/src/autofill/services/desktop-autofill-settings.service.ts:1]: The Duck Duck Go integration doesn't feel like an "autofill setting" to me. Could we make this service "DesktopSettings" (to represent a service whose concern is client-specific settings) instead?

I have no issue with this living as a Autofill team ownership/concern though, if that's desired.

**Review by jprusik — CHANGES_REQUESTED** [review]

File: apps/desktop/src/autofill/services/desktop-autofill-settings.service.ts
Comment (jprusik) [reply apps/desktop/src/autofill/services/desktop-autofill-settings.service.ts:1]: Thinking on this more - this probably belongs in a `DesktopIntegrationsService` with `enableBrowserIntegration` and `enableBrowserIntegrationFingerprint`.

**Review by jprusik — APPROVED** [review]
See previous feedback - will follow up offline. Otherwise LGTM

File: apps/desktop/src/app/accounts/settings.component.ts [hunk 2]
```diff
@@ -259,8 +261,9 @@ export class SettingsComponent implements OnInit {
       enableBrowserIntegration: await this.stateService.getEnableBrowserIntegration(),
       enableBrowserIntegrationFingerprint:
         await this.stateService.getEnableBrowserIntegrationFingerprint(),
-      enableDuckDuckGoBrowserIntegration:
-        await this.stateService.getEnableDuckDuckGoBrowserIntegration(),
```
Comment (MGibson1) [inline apps/desktop/src/app/accounts/settings.component.ts:263]: **issue**: Please remove this and the set methods from state service and the value from global state

**Review by MGibson1 — CHANGES_REQUESTED** [review]

**Review by jprusik — DISMISSED** [review]

**Review by jprusik — DISMISSED** [review]

**Review by MGibson1 — APPROVED** [review]

**Review by cagonzalezcs — APPROVED** [review]

**Pull request merged**